packages/app/src/components/VisualizationTypeSelector/VisualizationTypeSelector.js

```diff
@@ -38,6 +38,15 @@ export const VisualizationTypeSelector = (
 
     const handleListItemClick = type => () => {
         setUi(getAdaptedUiByType({ ...ui, type }))
+        setUiSeries(
+            uiSeries.map(item => {
+                const tempItem = { ...item }
+                if (tempItem.type) {
+                    delete tempItem.type
+                }
+                return tempItem
+            })
```

Comment:
You can extract this to the `setUiSeries` function itself by accessing `getState` in `mapDispatchToProps` (example below). Then you don't have to pass `uiSeries` as a prop to the component just to send it back.

Example: https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/components/UpdateButton/UpdateVisualizationContainer.js#L59

Reply:
Good point, fixed